torch/_decomp/decompositions.py

```diff
@@ -760,9 +760,9 @@ def embedding_dense_backward(
     scale_grad_by_freq: bool,
 ):
     numel = indices.numel()
-    grad = grad_output.reshape(numel, grad_output.size(-1))
+    grad = grad_output.reshape((numel, grad_output.size(-1)))
     grad_weight = grad_output.new_zeros((num_weights, grad_output.shape[-1]))
-    indices_rank1 = indices.reshape(numel)
+    indices_rank1 = indices.flatten()
```

Comment:
@ezyang this change (and the one above) are fine, decomps previously assumed indices are contiguous (otherwise you can't call reshape), but it's not guaranteed.

Reply:
This was same issue, at the time `reshape` did not accept unpacked shapes, so it had to be `indices.reshape((numel))` and it just felt cleaner to use `flatten()` instead which is equivalent.